src/content/blog/2025_ucc_unitaryhack.md

```diff
@@ -0,0 +1,136 @@
+
+**The experience was deeply instructive**, and I was fortunate to contribute to the project, though it was UCC’s benchmarking approach that uncovered the limitation in the first place—a contribution I remain genuinely thankful for.
+Additionally, UCC’s infrastructure emphasized the importance of compatibility between approximate and exact compilation methods. To support this, I introduced a fallback mechanism that checks fidelity and cost metrics, defaulting to UCC’s exact compiler when approximation underperforms. It was a straightforward solution, but in hindsight, one that proved to be quite practical.
+
+**Future projects I'm interested in working on:**
+My current focus is on finishing [qmprs](https://github.com/Qualition/qmprs). Given the approximate compilation module making it to production, I will be looking forward to integrating qmprs once it reaches a minimum mature stage in UCC with the help of the supportive maintainers.
+Feel free to reach out to me via LinkedIn, or discord, or email should you have any questions or would like to collaborate.
+
+--
+
+#### Porting over a BQSKiT compiler pass
+[GH issue](https://github.com/unitaryfoundation/ucc/issues/380)  
+Bounty: $500  
+Closed by: [WolfLink](https://github.com/WolfLink)  
+
+Hi, I'm Marc Davis, currently pursuing a PhD at MIT researching quantum compiling techniques.
+
+**What drew me to UCC:**
+In the course of my research, I work with a variety of quantum circuit compilation, transpilation, and optimization tools, and a common problem is interfacing between different tools, which often rely on different underlying circuit formats, sometimes with different sets of features. UCC's goal of making a wrapper that allows the interplay of many different tools is a worthy one.
+
+<div style="display: flex; flex-direction: column; align-items: center; margin: 20px 0;">
+  <img src="/images/2025_ucc_unitaryhack/bqskit-logo.png" alt="BQSKit logo" width="200" height="auto" />
+  <span style="font-size: 0.9em; color: #666; margin-top: 4px; display: block; text-align: center;">
+    Source: <a href="https://bqskit.readthedocs.io/" target="_blank" style="color: #666; text-decoration: underline;">BQSKit: Berkeley Quantum Synthesis Toolkit</a>
+  </span>
+</div>
+
+**Challenges I faced:**
+My idea was to combine UCC and BQSKit. BQSKit is a quantum compiling tool I have previously worked on that provides a suite of optimization techniques. By implementing BQSKit as a UCC pass, I enabled many powerful circuit optimization tools to be used with the UCC ecosystem. However, many of BQSKit's tools are focused on providing highly efficient circuits at the cost of runtime on the order of hours. UCC has strict runtime limits on the order of less than a second. A compromise was reached, to make BQSkit an optional pass rather than a core part of the UCC workflow.
```

Comment:
This feels less like a challenge and more like a statement about the priorities of the projects. I think it's very much worth calling out, but maybe it can be framed as something like "trying to merge two compilers whose underlying goals are different in a way that makes users of UCC happy". And the solution was a design decision, not a technical one. WDYT?